### ISSUE TITLE
Got poor model performance with gradient_accumulation_steps=8

### ISSUE THREAD

**Issue by ShuGao0810** [llamazing/numnet_plus#5]
The model performance training with gradient_accumulation_steps=8 is much worse than expeced, but why? All hyperparameters are consistent with yours, except gradient_accumulation_steps, set as 8 for gpu memory consideration。

train batch size=16
gradient_accumulation_steps=8

+ python drop_eval.py --gold_path drop_dataset/drop_dataset_dev.json --prediction_path prediction.json
Exact-match accuracy 60.57
F1 score 64.48
60.57   &   64.48
----
date: 147 (1.54%)
 Exact-match accuracy 27.211
 F1 score 32.129
number: 5911 (61.99%)
 Exact-match accuracy 65.353
 F1 score 65.549
span: 3010 (31.56%)
 Exact-match accuracy 59.336
 F1 score 66.323
spans: 468 (4.91%)
 Exact-match accuracy 18.590
 F1 score 49.218

**Comment by llamazing** [comment]
We have update training hyperparameters in README.
Using new hyperparameters you can get `EM: 78 && F1 82` with  `batch size=16 &&  gradient_accumulation_steps=4`.
#3 also got pool result using `gradient_accumulation_steps=8`.
It seems that the `gradient_accumulation_steps ` affects training results greatly.

**Comment by unbreading** [comment]
Hi, @ShuGao0810. 
I got the result 
`Exact-match accuracy 78.69
F1 score 82.15`
with `gradient_accumulation_steps=16.`
`8` could be a magic number for parameter `gradient_accumulation_steps` (facepalm

Hi, @llamazing. 
May I ask what's the differences between NumNet+ and NumNet+ v2? I'm curious about it.
Are commits on Oct 28&29 the updates for v2? Thanks.

**Comment by llamazing** [comment]
> Hi, @ShuGao0810.
> I got the result
> `Exact-match accuracy 78.69 F1 score 82.15`
> with `gradient_accumulation_steps=16.`
> `8` could be a magic number for parameter `gradient_accumulation_steps` (facepalm
> 
> Hi, @llamazing.
> May I ask what's the differences between NumNet+ and NumNet+ v2? I'm curious about it.
> Are commits on Oct 28&29 the updates for v2? Thanks.

NmNet+ v2 uses tag-based multi-span extraction which gets a better result on multi-span questions.
